src/main/scala/com/fulcrumgenomics/vcf/filtration/FilterSomaticVcf.scala

```diff
@@ -182,16 +183,21 @@ import scala.collection.immutable.ListMap
         case Nil     => variant
         case filters =>
           val pileup         = piler.pileup(refName = variant.chrom, pos = variant.pos).withoutOverlaps
-          val allAnnotations = variant.attrs.toBuffer
-          val allFilters     = variant.filters.toBuffer
+          val newAnnotations = new ListBuffer[(String, Any)]
+          val newFilters     = new ListBuffer[String]
 
           filters.foreach { filter =>
             val annotations = filter.annotations(pileup, variant.genotypes(name))
-            allAnnotations ++= annotations
-            allFilters     ++= filter.filters(annotations)
+            newAnnotations ++= annotations
+            newFilters     ++= filter.filters(annotations)
+          }
+
+          val allFilters = {
+            if (variant.filters == PassingFilters && newFilters.nonEmpty) newFilters.toSet
+            else variant.filters ++ newFilters
```

Comment:
Appending filters to a `PASS` in a variant seems like an easy mistake (which I've made!)

What do you think of a helper that would perform a copy and handle `PASS` correctly like:

- `Variant.withFilters()`
- `Variant.withFilter()`

The plural would add all filters in a single copy. The singular would just add one filter.